### PR TITLE
CXX: Improve namespace handling, add support for namespace aliases.

### PR DESCRIPTION
**Pull request by pragmaware**
- Fixes issue #992
- Adds support for namespace alises
- Adds support for inline namespaces (marked in properties)

### CODE REVIEW

File: Units/parser-cxx.r/namespace.cpp.d/input.cpp [hunk 1]
```diff
@@ -1,10 +1,55 @@
...
+}
+
+namespace e1 = a1::a2::a3;
+
+namespace e2 = b1::b2 __attribute__((abi_tag("blah")));
+
+namespace z1 { };
+
+
+
```
Comment (masatake) [inline Units/parser-cxx.r/namespace.cpp.d/input.cpp]: These lines are not needed.

**Comment by masatake** [comment]
Thank you.
Many unnecessary whitespaces are introduced. Could you trim them? You can find them with `git diff`.

File: parsers/cxx/cxx_parser_namespace.c [hunk 2]
```diff
@@ -54,105 +55,234 @@ boolean cxxParserParseNamespace(void)
...
+
 		if(!cxxParserParseNextToken())
 		{
 			// syntax error, but we tolerate this
-			CXX_DEBUG_LEAVE_TEXT("Implicit EOF in cxxParserParseNextToken");
+			CXX_DEBUG_LEAVE_TEXT("EOF in cxxParserParseNextToken");
 			return TRUE; // EOF
 		}
-
+
```
Comment (masatake) [inline parsers/cxx/cxx_parser_namespace.c]: This is an example of whitespace, which is not needed.
Before making a PR, please, check your change with git diff. The white spaces are printed with red in git diff output.
See https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#Formatting-and-Whitespace

**Comment by pragmaware** [comment]
Ops... merged before you actually reviewed it as I wanted to work on __attribute__ which involves modifying the same files. It was waiting since some time and I've thought it was OK.
Anyway, I'll trim whitespaces in a commit attached to pull #1005.

**Comment by masatake** [comment]
I'm sorry for taking time. I have small troubles in my health.

I'm reworking on --field option now. During it I found critical issue about these commits.
You intrdouced "name" field for C++. Please, change the name. It is already used as a common field name.
```
% ./ctags --list-fields | grep 'name'
N       name            on      NONE             TRUE     tag name (fixed field)        
K       NONE            off     NONE             TRUE     Kind of tag as full name      
t       typeref         on      NONE             TRUE     Type and name of a variable or typedef
Z       scope           off     NONE             TRUE     Include the "scope:" key in scope field (use s) in tags output, scope name in xref output
p       scopeKind       on      NONE             TRUE     Kind of scope as full name    
-       name            on      C++              TRUE     aliased names                 
```

**Pull request merged**